### PR TITLE
Add package py-guiqwt and dependencies

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-qtconsole/package.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    depends_on('py-setuptools',          type='build')
+    depends_on('py-ipykernel@4.1:',      type=('build', 'run'))
+    depends_on('py-jupyter-client@4.1:', type=('build', 'run'))
+    depends_on('py-jupyter-core',        type=('build', 'run'))
+    depends_on('py-pygments',            type=('build', 'run'))
+    depends_on('py-traitlets',           type=('build', 'run'))
+    depends_on('py-sphinx@1.3:',         type=('build', 'run'), when='+docs')
+    #mock; python_version=="2.7" and extra == 'test'
+    #pexpect; sys_platform != "win32" and extra == 'test'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-qtconsole/package.py]: These will cause the flake8 tests to fail. Comments have to have a space after them.

The test dependency on mock will have to wait until #2869 and #1279 are complete.
I don't think we need to worry about the Windows dependency. 😆

File: var/spack/repos/builtin/packages/py-guidata/package.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    homepage = "https://github.com/PierreRaybaut/guidata"
+    url      = "https://pypi.io/packages/source/g/guidata/guidata-1.7.5.zip"
+
+    version('1.7.5', '915188c02ad3c89951ee260db65d84a7')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-pyqt@4:')
+    depends_on('py-spyder@2.0:2.9.9')
+    depends_on('py-h5py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-guidata/package.py]: `type=('build', 'run')`

**Comment by adamjstewart** [comment]
@citibeth can you approve?

**Comment by healther** [comment]
ping

**Comment by healther** [comment]
@citibeth do you want additional changes? I would really prefer to have this PR merged

**Pull request merged**